bitcoin/src/crypto/key.rs

```diff
@@ -700,6 +701,7 @@ pub struct TweakedPublicKey(XOnlyPublicKey);
 impl fmt::LowerHex for TweakedPublicKey {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::LowerHex::fmt(&self.0, f) }
 }
+impl_to_hex_from_lower_hex!(TweakedPublicKey, |_| size_of::<TweakedPublicKey>());
```

Comment:
I think we should use serialized size, not `size_of` since There could reasonably be some optimizations that keep the key larger in memory to get improved performance on crypto operations. I guess this applies to other types as well.

Reply:
Agreed. `size_of` is semantically wrong. In cases like keys it would be better to just use magic numbers than `size_of`. Though I'm sure we have constants somewhere you should use instead.

Reply:
By cases like keys, do you mean where the data can be serialized? Or should something like 
`impl_to_hex_from_lower_hex!(ServiceFlags, |_| size_of::<ServiceFlags>());` also be rewritten to 
`impl_to_hex_from_lower_hex!(ServiceFlags, |_| 8_usize);`

Reply:
Anywhere you are using `size_of` should be replaced. The output of `size_of` is just as magical as a hardcoded `8` (in fact, it is significantly more magical since the value is hard to determine and may be different on different systems).